src/NovelRT.Interop/LifetimeExtender.h

```diff
@@ -0,0 +1,57 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+#ifndef NOVELRT_INTEROP_LIFETIMEEXTENDER_H
+#define NOVELRT_INTEROP_LIFETIMEEXTENDER_H
+#include <memory>
+#include <vector>
+/*
+ * Extends the lifetime of objects contained in a shared_ptr.
+ * This lets the API consumer take ownership of an object without double indirection.
+ */
+namespace NovelRT::Interop::Lifetime
+{
+    template<typename T> std::vector<std::shared_ptr<T>>& AliveObjects()
+    {
+        static std::vector<std::shared_ptr<T>> aliveObjects{};
+
+        return aliveObjects;
+    }
+
+    template<typename T> void KeepAlive(std::shared_ptr<T>&& object)
+    {
+        AliveObjects<T>().emplace_back(std::move(object));
```

Comment:
I think you might need `std::forward` not `std::move` here.

Reply:
Isn't `std::forward` reserved for perfect forwarding? For cases where you have `T&&` in a template (and not `std::shared_ptr<T>&&`).

Here, `std::move` just conducts the moving operation, because `emplace_pack(object)` would still consider `object` as an l-value.

Reply:
Can't you perfect forward the pointer here? its a `T&&`